### PR TITLE
Refactor/pcf 5 fix autoprefixer

### PR DESCRIPTION
**Pull request by CuriousCactus**
Some tinkering to get the spell list to display nicely on small screens.
As a side effect I got browser prefixes working.
I'll add comments with reasoning.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,9 +38,7 @@
       "not op_mini all"
     ],
     "development": [
-      "last 1 chrome version",
-      "last 1 firefox version",
```
Comment (CuriousCactus) [inline package.json:41]: "last 1 firefox version" didn't seem to work for me, despite it being in the readme for the package...

File: src/compiled/theme.css [hunk 2]
```diff
@@ -1 +1 @@
...
...ght:10px;cursor:pointer}.filter-button-wrapper{flex:0 1 100%}input.search-input{display:flex;width:-webkit-fill-available;width:-moz-available;width:stretch;height:30px;border:2px solid;border-radius:25px;padding:10px;background:0 0;font-size:20px;margin-bottom:10px}input.search-input::-webkit-input-placeholder{font-family:'Century Gothic',CenturyGothic,AppleGothic,sans-serif;font-size:20px}input.search-input:-ms-input-placeholder{font-family:'Century Gothic',CenturyGothic,AppleGothic,sans-serif;font-size:20px}input.search-input::-ms-input-placeholder{font-family:'Century Gothic',CenturyGothic,AppleGothic,sans-serif;font-size:20px}input.search-input::placeholder{font-family:'Century Gothic',CenturyGothic,AppleGothic,sans-serif;font-size:20px}svg.item-remove-pin{position:absolute;top:18px;left:15px;cursor:pointer;z-index:1}.dndapp-main{align-items:center;display:flex;flex-direction:column;font-family:'Century Gothic',CenturyGothic,AppleGothic,sans-serif;justify-content:center;min-height:100vh;text-align:center;transition:.25s ease-in-out}body{margin:0}h1,h2,h3,h4,h5,h6{margin:0;padding:10px}ul{margin:5px 0;-webkit-padding-start:20px;padding-inline-start:20px}ul li{margin-bottom:5px}
```
Comment (CuriousCactus) [inline src/compiled/theme.css:1]: Should this be gitignored?

File: src/styles/_main.scss [hunk 3]
```diff
@@ -484,9 +485,10 @@
 
     span {
       overflow: hidden;
-      display: flex;
       margin-right: auto;
       margin-left: 35px;
+      text-overflow: ellipsis;
+      white-space: nowrap;
```
Comment (CuriousCactus) [inline src/styles/_main.scss:491]: For very long spell names, e.g. anything by Mordenkainen.

File: src/styles/_main.scss [hunk 4]
```diff
@@ -555,6 +557,7 @@
     margin-right: 35px;
     font-size: $medium-size;
     opacity: .85;
+    white-space: nowrap;
```
Comment (CuriousCactus) [inline src/styles/_main.scss:560]: Don't let e.g. L: 8 wrap onto two lines

File: src/styles/_main.scss [hunk 5]
```diff
@@ -625,7 +628,7 @@
       flex-direction: column;
       max-width: 600px;
       margin: auto;
-      position: fixed;
+      position: absolute;
```
Comment (CuriousCactus) [inline src/styles/_main.scss:631]: So you can scroll down the filters now there are more of them

File: src/styles/_main.scss [hunk 6]
```diff
@@ -666,7 +669,7 @@ input {
   &.search {
     &-input {
       display: flex;
-      width: -webkit-fill-available;
+      width: stretch;
```
Comment (CuriousCactus) [inline src/styles/_main.scss:672]: This line prompted all the refactoring to get browser prefixes working

File: src/styles/theme.scss [hunk 7]
```diff
@@ -0,0 +1,9 @@
+// Load Variables first, order matters.
+@import 'variables.scss';
+// Load extends.
+@import 'extends.scss';
+// Then the theme, colour styles including variables.
+@import 'themes/*.scss';
+// The rest.
+@import 'animations.scss';
+@import 'main.scss';
```
Comment (CuriousCactus) [inline src/styles/theme.scss:9]: This is basically _compile.scss, moved and renamed during the refactor.

File: gulpfile.js [hunk 8]
```diff
@@ -1,32 +1,27 @@
...
-    .pipe(concat('theme.css'))
-    .pipe(sass({
-      outputStyle: output,
-      precision: 10,
-      includePaths: ['./src/styles/'],
-    }))
+    .pipe(sass())
     .on('error', sass.logError)
-    .pipe(gulp.dest(cssDest))
+    .pipe(autoprefixer())
```
Comment (CuriousCactus) [inline gulpfile.js:16]: Did some jiggling the gulp pipeline in order to get autoprefixer working.

**Comment by prophet-kn** [comment]
apparently I cannot push to the fork to fix conflicts... fucking github

File: src/compiled/theme.css (hunk 2 above)
Comment (prophet-kn) [reply src/compiled/theme.css:1]: can be, yep

Comment (prophet-kn) [reply src/compiled/theme.css:1]: ah no wait, this one pref not :) thought it was in build files

**Comment by prophet-kn** [comment]
gonna merge it here https://github.com/prophet-kn/dnd_spells_book/pull/51, no worries, your commits are authored by you :)

**Pull request closed without merging**